markets/perps-market/subgraph/tests/handleAccountCreated.ts

```diff
@@ -1,19 +1,24 @@
-import { assert } from 'matchstick-as';
+import { assert, log } from 'matchstick-as';
 import { handleAccountCreated } from '../optimism-goerli';
 import { createAccountCreatedEvent } from './event-factories/createAccountCreatedEvent';
 
 export default function test(): void {
-  const accountCreatedEvent = createAccountCreatedEvent(
-    1,
-    '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266',
-    10_000,
-    10
-  );
+  assert.entityCount('Account', 0);
 
-  handleAccountCreated(accountCreatedEvent);
+  log.info('Should create a new record for the account', []);
+  const owner1 = '0x4200000000000000000000000000000000000000';
+  handleAccountCreated(createAccountCreatedEvent(1, owner1, 10_000, 10));
+  assert.entityCount('Account', 1);
+  assert.fieldEquals('Account', '1', 'id', '1');
+  assert.fieldEquals('Account', '1', 'accountId', '1');
+  assert.fieldEquals('Account', '1', 'owner', owner1);
 
+  // TODO: This is most likely not how it supposed to work!
+  log.info('Should update a record for another account', []);
```

Comment:
I dont think this test make sense. The create account soldity function would revert if you try to create an account that already exists.

I guess we could add validation to the subgraph to throw if it happens. But if we start coding defensively for things that shouldn't happen the subgraph code might start to explode in complexity... 🤔 So I think it's also fine to not trigger event in ways that they couldn't be triggered onchain

Reply:
call stack:
createAccount -> accountTokenModule.safeMint -> ERC721._mint

`ERC721._mint` reverts if the account exists https://github.com/Synthetixio/synthetix-v3/blob/main/utils/core-contracts/contracts/token/ERC721.sol#L202